### PR TITLE
Code Push PoC / Integration

### PR DESCRIPTION
**Pull request by juanchoperezj**
### Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

Integration of the Code Push tool with the current base code, following cases to perform OTA updates:
* ON_APP_START
* ON_APP_RESUME

---

#### Jira board reference

- N/A

---

#### Notes:

- The aim of the PR is to be reviewed but not to be merged on the base, so we need to discuss it in order to make that decision.
- The PR is pointed for two general flows, those could change depending on the app's needs.
- Info about the install modes: https://learn.microsoft.com/en-us/appcenter/distribution/codepush/rn-api-ref#installmode

---

#### Tasks:

- [x] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [ ] Tested on a real device
- [ ] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

#### Preview

Check update ON_APP_START with install mode: IMMEDIATE

[code push - on app start.webm](https://user-images.githubusercontent.com/17632690/210406785-9e98a3ff-2e0f-432f-b83b-11d47461d7c1.webm)

Check update ON_APP_RESUME with install mode ON_NEXT_RESUME after the app was in the background at least for 2 minutes.

https://user-images.githubusercontent.com/17632690/210407826-db469e6a-b6d0-426e-9593-f1b1d5aec2b3.mp4

**Note: Once the app goes into the background, you can fast-forward the video to see the update**

### CODE REVIEW

File: android/app/src/main/java/com/reactnativebase/MainApplication.java [hunk 1]
```diff
@@ -10,82 +13,95 @@
...
 public class MainApplication extends Application implements ReactApplication {
 
-  private final ReactNativeHost mReactNativeHost =
-      new ReactNativeHost(this) {
-        @Override
-        public boolean getUseDeveloperSupport() {
-          return BuildConfig.DEBUG;
-        }
+    private final ReactNativeHost mReactNativeHost =
+            new ReactNativeHost(this) {
```
Comment (juanchoperezj) [inline android/app/src/main/java/com/reactnativebase/MainApplication.java:25]: Same as before

File: src/features/code-push/hooks/useCodePush.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  };
+
+  const checkForUpdate = async () => {
+    try {
+      const remotePackage = await CodePush.checkForUpdate(
+        undefined,
+        ({ updateAppVersion, appVersion }: RemotePackageBinary) => {
+          if (updateAppVersion) {
+            return Alert.alert(
+              'Update available',
```
Comment (fernandatoledo) [inline src/features/code-push/hooks/useCodePush.ts:44]: As i18n has already been merged, WDYT about using localization for these strings??

**Review by fernandatoledo — COMMENTED** [review]
Thank you so much for adding this 🎉

**Comment by juanchoperezj** [comment]
As discussed in the last sync of the year, we are closing this PR and leaving it as a reference [here](https://www.notion.so/rootstrap/8c417cf18152437f9df984529d41320e?v=35b778723b984ef9b7a2246158e16748)

**Pull request closed without merging**